tests/e2e/specs/shopper/checkout-different-addresses.test.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import {
+	merchant,
+	openDocumentSettingsSidebar,
+	uiUnblocked,
+} from '@woocommerce/e2e-utils';
+
+import {
+	visitBlockPage,
+	selectBlockByName,
+	saveOrPublish,
+} from '@woocommerce/blocks-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getNormalPagePermalink,
+	shopper,
+	visitPostOfType,
+} from '../../../utils';
+
+const config = require( 'config' );
+
+const block = {
+	name: 'Checkout',
+};
+
+const simpleProductName = '128GB USB Stick';
+const pageName = 'Checkout Block';
+
+if ( process.env.WOOCOMMERCE_BLOCKS_PHASE < 2 )
+	// eslint-disable-next-line jest/no-focused-tests
+	test.only( `skipping ${ block.name } tests`, () => {} );
+
+describe( `${ block.name } Block (frontend)`, () => {
+	let productPermalink;
+
+	beforeAll( async () => {
+		// prevent CartCheckoutCompatibilityNotice from appearing
+		await page.evaluate( () => {
+			localStorage.setItem(
+				'wc-blocks_dismissed_compatibility_notices',
+				'["checkout"]'
+			);
+		} );
+		await merchant.login();
+
+		// Display all address fields.
+		await visitBlockPage( pageName );
+		await openDocumentSettingsSidebar();
+		await selectBlockByName(
+			'woocommerce/checkout-shipping-address-block'
+		);
+		await expect( page ).toClick( 'label', { text: 'Company' } );
+		await expect( page ).toClick( 'label', {
+			text: 'Apartment, suite, etc.',
+		} );
+		await expect( page ).toClick( 'label', { text: 'Phone' } );
+		await saveOrPublish();
+
+		// Get product page permalink.
+		await visitPostOfType( simpleProductName, 'product' );
+		productPermalink = await getNormalPagePermalink();
+
+		await merchant.logout();
+	} );
+
+	afterAll( async () => {
+		// empty cart from shortcode page
+		await shopper.goToCart();
+		await shopper.removeFromCart( simpleProductName );
+		await page.evaluate( () => {
+			localStorage.removeItem(
+				'wc-blocks_dismissed_compatibility_notices'
+			);
+		} );
```

Comment:
You will need to also undo the setting of the fields you are specifically enabling in the before all block, because they might interfere with other test suites.

```
await expect( page ).toClick( 'label', { text: 'Company' } );
		await expect( page ).toClick( 'label', {
			text: 'Apartment, suite, etc.',
		} );
		await expect( page ).toClick( 'label', { text: 'Phone' } );
```

Reply:
> > Based on this information, would you still suggest sticking to the `frontend` and `backend` folders, risking that we might have to rename them later if WooCommerce Blocks gets finally merged into core?
> 
> I think we could do the renaming in this PR so that we have a shared understanding of the structure.

I thought better about this and I propose to do the renaming of the folders in another PR. This way we keep this PR clean and we have a separate refactoring PR.

So `frontend` -> `shopper` and `backend` -> `merchant`. We can make a separate ticket for it for better visibility, and also explaining the reasoning behind it.